README.md

```diff
@@ -3,11 +3,61 @@
 > :warning: **Under Development**
 
 [![Go Report Card](https://goreportcard.com/badge/github.com/openshift/managed-upgrade-operator)](https://goreportcard.com/report/github.com/openshift/managed-upgrade-operator)
-![Go Coverage](./coverage_badge.png)
+[![codecov](https://codecov.io/gh/openshift/managed-upgrade-operator/branch/master/graph/badge.svg)](https://codecov.io/gh/openshift/managed-upgrade-operator)
 [![GoDoc](https://godoc.org/github.com/openshift/managed-upgrade-operator?status.svg)](https://pkg.go.dev/mod/github.com/openshift/managed-upgrade-operator)
 [![License](https://img.shields.io/:license-apache-blue.svg)](http://www.apache.org/licenses/LICENSE-2.0.html)
 
+----
+
+The _Managed Upgrade Operator_ has been created for the [OpenShift Dedicated Platform](https://docs.openshift.com/dedicated/4/) (OSD) to manage the orchestration of automated in-place cluster upgrades.
+
+The operator's goal is to satisfy the operating conditions that a managed cluster must hold pre/post-invocation of a cluster upgrade. This may include, but is not limited to, actions such as:
+* Pre and post-upgrade health checks
+* Worker capacity scaling during the upgrade period
+* Maintenance window scheduling and notification
+
+Whilst the operator's job is to invoke a cluster upgrade, it does not perform any activities of the cluster upgrade process itself. This remains the responsibility of the OpenShift Container Platform.
+
+If you like to contribute to the Managed Upgrade Operator, please read our [Contribution Policy](./docs/contributing.md) first.
+
+----
+
+* [Info](#info)
+   * [Documentation](#documentation)
+      * [For Developers](#for-developers)
+   * [Workflow - UpgradeConfig](#workflow---upgradeconfig)
+      * [Example CR](#example-input-custom-resource)
+      
+# Info
+
+## Documentation
+
 ### For Developers
 
-* [Development](./docs/development.md) -- Instructions for developing the operator. 
+* [Design](./docs/design.md) -- Describes the interaction between the operator and the custom resource definition.
+* [Development](./docs/development.md) -- Instructions for developing and deploying the operator. 
 * [Testing](./docs/testing.md) -- Instructions for writing tests.
+
+## Workflow - UpgradeConfig
+
+1. The operator watches all namespaces for `UpgradeConfig` resources.
```

Comment:
It's not the place for this, I know, but it got me thinking we should only watch on 1 resource. CVO creates their ClusterVersion resource and only reacts to that. It might be worth doing the same with UpgradeConfig. managed-upgrade-operator? create an initial resource that matches CVO?

Reply:
Yeah I'd agree on that one, in that there should only need to be 1 resource. I'm going to reword this in advance of us maybe carding work around it.